src/k8s/pkg/snap/util/services.go

```diff
@@ -75,3 +75,20 @@ func StopK8sDqliteServices(ctx context.Context, snap snap.Snap) error {
 	}
 	return nil
 }
+
+// ServiceArgsFromMap processes a map of string pointers and categorizes them into update and delete lists.
+// - If the value pointer is nil, it adds the argument name to the delete list.
+// - If the value pointer is not nil, it adds the argument and its value to the update map.
+func ServiceArgsFromMap(args map[string]*string) (map[string]string, []string) {
```

Comment:
`snaputil` is reserved for utilities that interact with snaps. This is not one of them, so let's move it to `pkg/utils`